### DISCUSSION TITLE
Memory leak in flatpak helper programs

### DISCUSSION THREAD

**Discussion by jfbauer432** [StreamController/StreamController#217] [General]
I noticed a couple of programs that are increasing their memory usage a lot when StreamController is running: xdg-dbus-proxy & flatpak-session-helper.  CPU time isn't trivial either.  This is possibly related to a known issue with flatpak (see below).

There may not be anything StreamController can do here (which is why I posted here instead of in issues).  But I am posting this in case anyone has an idea of what might be triggering the memory usage and what , if any, config change can be made to reduce the impact.

Here is part of top output after running for about 5 hours with not doing anything with StreamController.  Running with a custom flatpak version that includes commit 57431f9.

```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                              
  85083 jfb       20   0  914496 683904   5120 S   4.6   8.4  17:02.03 xdg-dbus-proxy --args=42             
  85074 jfb       20   0  811404 522368   6400 R  14.5   6.4  29:45.24 /usr/libexec/flatpak-session-helper  
  85091 jfb       20   0 3801852 295408 142540 S   3.0   3.6  13:22.64 StreamController                     
```
xdg-dbus-proxy goes away when quitting StreamController.  But flatpak-session-helper stays running and its memory consumption stops going up (but doesn't drop).  Killing xdg-dbus-proxy while StreamController is running, kills StreamController.

There are known (and fixed but not released) memory leak issues with flatpak as indicated in the release notes for [1.15.10 prerelease](https://github.com/flatpak/flatpak/releases) version that may or may not address this.  I have not looked into how to update flatpak with a pre-release version.

**Comment by Core447** [comment]
> There may not be anything StreamController can do here (which is why I posted here instead of in issues)

Yeah if the described memory leaks are really caused by Flatpak itself, there's nothing I can do about it, sorry.
